resources/User.py

```diff
@@ -1,69 +1,46 @@
-from werkzeug.security import generate_password_hash, check_password_hash
+from werkzeug.security import generate_password_hash
 from flask_restful import Resource
-from flask import request, jsonify
-import uuid
-import os
-import jwt
+from flask import request
+from middleware.user_queries import user_post_results
 
 
 class User(Resource):
     def __init__(self, **kwargs):
         self.psycopg2_connection = kwargs["psycopg2_connection"]
 
-    # Login function: allows a user to login using their email and password as credentials
-    # The password is compared to the hashed password stored in the users table
-    # Once the password is verified, an API key is generated, which is stored in the users table and sent to the verified user
-    def get(self):
+    def post(self):
+        """
+        Sign up function: allows a user to sign up by submitting an email and password.
+        The email and a hashed password are stored in the users table and this data is returned to the user upon completion
+        """
         try:
             data = request.get_json()
             email = data.get("email")
             password = data.get("password")
             cursor = self.psycopg2_connection.cursor()
-            cursor.execute(
-                f"select id, password_digest from users where email = '{email}'"
-            )
-            results = cursor.fetchall()
-            user_data = {}
-            if len(results) > 0:
-                user_data = {"id": results[0][0], "password_digest": results[0][1]}
-            else:
-                return {
-                    "message": "There username or password is incorrect. Please try again."
-                }, 400
+            user_post_results(cursor, email, password)
+            self.psycopg2_connection.commit()
 
-            if check_password_hash(user_data["password_digest"], password):
-                api_key = uuid.uuid4().hex
-                user_id = str(user_data["id"])
-                cursor.execute(
-                    "UPDATE users SET api_key = %s WHERE id = %s", (api_key, user_id)
-                )
-                payload = {
-                    "message": "API key successfully created",
-                    "api_key": api_key,
-                }
-                self.psycopg2_connection.commit()
-                return payload
+            return {"message": "Successfully added user"}
 
         except Exception as e:
             self.psycopg2_connection.rollback()
             print(str(e))
-            return {"message": str(e)}, 500
+            return {"message": e}, 500
 
-    # Sign up function: allows a user to sign up by submitting an email and password. The email and a hashed password are stored in the users table and this data is returned to the user upon completion
-    def post(self):
+    # Endpoint for updating a user's password
+    def put(self):
         try:
             data = request.get_json()
             email = data.get("email")
```

Comment:
@mbodeantor Since this is an authenticated route on the client (as opposed to the reset route, which is unauthenticated), can we check the access token here (passed as the `Authorization` header) before changing the PW?